source/grid/tria.cc

```diff
@@ -4022,12 +4022,6 @@ namespace internal
                 // user flag is set or not
                 cell->clear_user_flag();
 
-                // An assert to make sure that the static_cast in the
-                // next line has the chance to give reasonable
-                // results.
-                Assert(cell->material_id()<= std::numeric_limits<types::material_id>::max(),
-                       ExcIndexRange(cell->material_id(),0,std::numeric_limits<types::material_id>::max()));
-
```

Comment:
I cannot figure out the point of this assertion. How can a value be greater than the maximum value for its type?

Reply:
It can't, of course. That's what the warning says.

Your question is probably what the person writing the code may have thought. I think for that you have to recall that we didn't always use these typedefs like `types::material_id`, but that the function may have returned an integer but we internally only stored a `char`, or the other way around.